streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -172,7 +172,7 @@
      * </pre>
      */
     public enum State {
-        CREATED(1, 2, 3), RUNNING(2, 3), REBALANCING(1, 2, 3), PENDING_SHUTDOWN(4), NOT_RUNNING;
+        CREATED(1, 2, 3), REBALANCING(1, 2, 3), RUNNING(1, 2, 3), PENDING_SHUTDOWN(4), NOT_RUNNING;
```

Comment:
I think it is a timing issue: when all the threads of an instance start at the same time, there could be a short period of time when the first rebalance get triggered, so the transition is `created -> running -> rebalancing -> running ..`; but if some threads start later than other, then the instance may be not in the `running` state before coming to the `rebalancing` state, so the transition is `created -> rebalancing -> running .. `.